scripts/Engineering/EnggUtils.py

```diff
@@ -116,20 +130,448 @@ def default_ceria_expected_peaks(final=False):
+    van_ws_foc_rb = mantid.RebinToWorkspace(WorkspaceToRebin=van_ws_foc, WorkspaceToMatch=sample_ws_foc,
+                                            OutputWorkspace=VAN_CURVE_REBINNED_NAME)  # copy so as not to lose data
+    sample_ws_foc = mantid.Divide(LHSWorkspace=sample_ws_foc, RHSWorkspace=van_ws_foc_rb,
+                                  OutputWorkspace=sample_ws_foc.name(), AllowDifferentNumberSpectra=False)
+    sample_ws_foc = mantid.ReplaceSpecialValues(InputWorkspace=sample_ws_foc, OutputWorkspace=sample_ws_foc.name(),
+                                                NaNValue=0, NaNError=0.0, InfinityValue=0, InfinityError=0.0)
+    return sample_ws_foc
+
+
+def _save_output_files(focus_dirs, sample_ws_foc, calibration, van_run, rb_num=None):
+    # set bankid for use in fit tab
+    foc_suffix = calibration.get_foc_ws_suffix()
+    xunit = sample_ws_foc.getDimension(0).name
+    xunit_suffix = XUNIT_SUFFIXES[xunit]
+    sample_run_no = sample_ws_foc.run().get('run_number').value
+    # save all spectra to single ASCII files
+    ascii_fname = _generate_output_file_name(calibration.get_instrument(), sample_run_no, van_run,
+                                             calibration.get_group_suffix(), xunit_suffix, ext='')
+
+    for focus_dir in focus_dirs:
+        if not path.exists(focus_dir):
+            makedirs(focus_dir)
+        mantid.SaveGSS(InputWorkspace=sample_ws_foc, Filename=path.join(focus_dir, ascii_fname + '.gss'),
+                       SplitFiles=False, UseSpectrumNumberAsBankID=True)
+        mantid.SaveFocusedXYE(InputWorkspace=sample_ws_foc, Filename=path.join(focus_dir, ascii_fname + ".abc"),
+                              SplitFiles=False, Format="TOPAS")
+        # Save nxs per spectrum
+        nxs_paths = []
+        mantid.AddSampleLog(Workspace=sample_ws_foc, LogName="Vanadium Run", LogText=van_run)
+        for ispec in range(sample_ws_foc.getNumberHistograms()):
+            # add a bankid and vanadium to log that is read by fitting model
+            bankid = foc_suffix if sample_ws_foc.getNumberHistograms() == 1 else f'{foc_suffix}_{ispec + 1}'
+            mantid.AddSampleLog(Workspace=sample_ws_foc, LogName="bankid",
+                                LogText=bankid.replace('_', ' '))  # overwrites
+            # save spectrum as nexus
+            filename = _generate_output_file_name(calibration.get_instrument(), sample_run_no, van_run, bankid,
+                                                  xunit_suffix, ext=".nxs")
+            nxs_path = path.join(focus_dir, filename)
+            mantid.SaveNexus(InputWorkspace=sample_ws_foc, Filename=nxs_path, WorkspaceIndexList=[ispec])
+            nxs_paths.append(nxs_path)
+    return nxs_paths  # from last focus_dir only
+
+
+def _generate_output_file_name(inst, sample_run_no, van_run_no, suffix, xunit, ext=""):
+    return "_".join([inst, sample_run_no, van_run_no, suffix, xunit]) + ext
+
+
+# DEPRECATED FUNCTIONS BELOW
```

Comment:
Good job on separating these out. Will make it easier to delete stuff when remove the deprecated algorithms